smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CommandGenerator.java

```diff
@@ -172,32 +177,16 @@ private void generateCommandConstructor() {
         });
     }
 
-    private void generateCommandMiddlewareResolver(String configType) {
-        Symbol serde = TypeScriptDependency.MIDDLEWARE_SERDE.createSymbol("getSerdePlugin");
-        writer.writeDocs("@internal");
-        writer.write("resolveMiddleware(")
-                .indent()
-                .write("clientStack: MiddlewareStack<$L, $L>,", "ServiceInputTypes", "ServiceOutputTypes")
-                .write("configuration: $L,", configType)
-                .write("options?: $T", applicationProtocol.getOptionsType())
-                .dedent();
-        writer.openBlock("): Handler<$T, $T> {", "}", inputType, outputType, () -> {
-            // Add serialization and deserialization plugin.
-            writer.write("this.middlewareStack.use($T(configuration, this.serialize, this.deserialize));", serde);
-
-            // Add customizations.
-            addCommandSpecificPlugins();
-
-            // EndpointsV2
-            if (service.hasTrait(EndpointRuleSetTrait.class)) {
-                writer.addImport(
-                    "getEndpointPlugin",
-                    "getEndpointPlugin",
-                    "@aws-sdk/middleware-endpoint"
-                );
+    private void generateEndpointParameterInstructionProvider() {
+        if (!service.hasTrait(EndpointRuleSetTrait.class)) {
+            return;
+        }
+        writer.addImport("EndpointParameterInstructions", null, "@aws-sdk/middleware-endpoint");
+        writer.openBlock(
+            "public static instructions(): EndpointParameterInstructions {", "}",
```

Comment:
Be more specific, like `getEndpointParameterInstructions`?
```suggestion
            "public static getEndpointParameterInstructions(): EndpointParameterInstructions {", "}",
```